### PR TITLE
fix: Fix Java helm charts to work with refactored logic. Fix FTS image

### PR DESCRIPTION
**Pull request by adchia**
**What this PR does / why we need it**:
This adds gcp/aws deps to the feature transformation server image as well, and also improves docs for dev experience around helm charts and Feast Java

It also creates templates for the READMEs so they generate the right "required" docs from using `helm-docs`
**Which issue(s) this PR fixes**:

Fixes #

### CODE REVIEW

File: examples/java-demo/feature_repo/application-override.yaml [hunk 1]
```diff
@@ -9,10 +9,19 @@ feature-server:
               config:
                 host: my-redis-master
                 port: 6379
-                password: [YOUR PASSWORD]
+                password: YTVWvjrQcK
```
Comment (felixwang9817) [inline examples/java-demo/feature_repo/application-override.yaml:12]: is this safe?

File: infra/charts/feast-python-server/README.md [hunk 2]
```diff
@@ -17,8 +17,8 @@ Feast Feature Server in Python
 | affinity | object | `{}` |  |
 | fullnameOverride | string | `""` |  |
 | image.pullPolicy | string | `"IfNotPresent"` |  |
-| image.repository | string | `""` | [required] The repository for the Docker image |
```
Comment (felixwang9817) [inline infra/charts/feast-python-server/README.md:20]: why is this no longer required?

File: infra/charts/feast-python-server/README.md [hunk 3]
```diff
@@ -36,24 +36,4 @@ Feast Feature Server in Python
 | tolerations | list | `[]` |  |
 
 ----------------------------------------------
-Autogenerated from chart metadata using [helm-docs v1.5.0](https://github.com/norwoodj/helm-docs/releases/v1.5.0)
-
-
-Docker repository and tag are required. Helm install example:
```
Comment (felixwang9817) [inline infra/charts/feast-python-server/README.md:42]: was it deliberate to delete this?

**Review by felixwang9817 — COMMENTED** [review]
some qs

File: examples/java-demo/feature_repo/application-override.yaml (hunk 1 above)
Comment (achals) [inline examples/java-demo/feature_repo/application-override.yaml:12]: ?

File: examples/java-demo/feature_repo/application-override.yaml [hunk 4]
```diff
@@ -9,10 +9,19 @@ feature-server:
...
             entityKeySerializationVersion: 2
+    # Uncomment below for dev
+    image:
+      tag: dev
+
+# Uncomment below for dev
+transformation-service:
+  image:
+    tag: dev
+
```
Comment (achals) [inline examples/java-demo/feature_repo/application-override.yaml:22]: should this be commented?

File: examples/java-demo/feature_repo/application-override.yaml [hunk 5]
```diff
@@ -9,10 +9,19 @@ feature-server:
...
 global:
   registry:
-    path: gs://[YOUR BUCKET]/demo-repo/registry.db
+    path: gs://danny-test-feast-bucket/demo-repo/registry.db
```
Comment (achals) [inline examples/java-demo/feature_repo/application-override.yaml:25]: ?

File: examples/java-demo/feature_repo/feature_store.yaml [hunk 6]
```diff
@@ -1,9 +1,9 @@
-registry: gs://[YOUR BUCKET]/demo-repo/registry.db
+registry: gs://danny-test-feast-bucket/demo-repo/registry.db
 project: feast_java_demo
 provider: gcp
 online_store:
   type: redis
-  connection_string: localhost:6379,password=[YOUR PASSWORD]
+  connection_string: localhost:6379,password=YTVWvjrQcK
```
Comment (achals) [inline examples/java-demo/feature_repo/feature_store.yaml:6]: ?

**Comment by adchia** [comment]
@achals @felixwang9817 PTAL

**Review by achals — APPROVED** [review]
/lgtm

**Pull request merged**